cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/SearchController.java

```diff
@@ -0,0 +1,61 @@
+package edu.asu.diging.cord19.explorer.web;
+
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.cord19.explorer.core.model.impl.PublicationImpl;
+import edu.asu.diging.cord19.explorer.core.mongo.impl.AffiliationPaperAggregationOutput;
+import edu.asu.diging.cord19.explorer.core.mongo.impl.SearchProvider;
+import edu.asu.diging.cord19.explorer.web.model.SearchType;
+
+@Controller
+public class SearchController {
+
+    @Autowired
+    private Map<SearchType, SearchProvider> searchProviderRegistry;
+
+    @RequestMapping(value = "/{searchType}/search")
+    public String search(@PathVariable("searchType") SearchType searchType, @RequestParam("search") String query,
```

Comment:
does this work if searchType is affiliations not AFFILIATIONS? If not, I would read this into a string and then do the conversion in the method. That would also give you a better error handling.